.buildkite/steps/publish-debian-package.sh

```diff
@@ -1,21 +1,36 @@
 #!/bin/bash
-set -e
+set -euo pipefail
 
 artifacts_build=$(buildkite-agent meta-data get "agent-artifacts-build" )
 
-dry_run() {
-  if [[ "${DRY_RUN:-}" == "false" ]] ; then
-    "$@"
-  else
-    echo "[dry-run] $*"
-  fi
-}
+# /root/,gnupg is a tmpfs volume, so we can safely store key data there and know
+# it won't be written to a disk
+secret_key_path="/root/.gnupg/gpg-secret.gpg"
+public_key_path="/root/.gnupg/gpg-public.gpg"
```

Comment:
Maybe we could assert tmpfs here so it can't accidentally write to disk if the docker invocation changes?

Reply:
Do you have a recommended approach for the assertion? Check the results of `mount | grep gnupg`?

Reply:
As we discovered while pairing:

```shell
apk add --update findmnt

findmnt --source tmpfs --target /root/.gnupg
```

Which will show a nice summary on success, and exit non-zero on failure.

```
TARGET SOURCE FSTYPE OPTIONS
/hello tmpfs  tmpfs  rw,nosuid,nodev,noexec,relatime
```